lib/grape-swagger/endpoint.rb

```diff
@@ -282,14 +282,25 @@ def default_type(params)
       end
     end
 
-    def parse_request_params(params)
-      array_key = nil
+    def parse_request_params(params, settings)
+      array_keys = []
       params.select { |param| public_parameter?(param) }.each_with_object({}) do |param, memo|
         name, options = *param
         param_type = options[:type]
         param_type = param_type.to_s unless param_type.nil?
-        array_key = name.to_s if param_type_is_array?(param_type)
-        options[:is_array] = true if array_key && name.start_with?(array_key)
+
+        array_keys << name.to_s if param_type_is_array?(param_type)
```

Comment:
Previously it was overriting `array_key` which could lead to the problem with such nesting:
```ruby
optional :a do
  optional :b do
    optional :c
  end
  optional :d
end
```
Previously here `:d` would not be detected as an array type